zio-http/src/main/scala/zhttp/http/Middleware.scala

```diff
@@ -237,6 +238,22 @@ object Middleware {
         } yield Patch.empty
     }
 
+  /**
+   * Creates a middleware for signing cookies
+   */
+  def signCookies[R, E](secret: String): Middleware[R, E] =
+    patch((_, _) =>
+      Patch.updateHeaders(h => {
+        Headers(
+          h.toList.collect {
+            case h if contentEqualsIgnoreCase(h._1, HeaderNames.setCookie) => {
+              (HeaderNames.setCookie, Cookie.decodeResponseCookie(h._2.toString).get.sign(secret).encode)
+            }
+          },
+        )
+      }),
+    )
```

Comment:
we can add a new operator inside of `Middleware` called `modifyHeaders`

```scala
def modifyHeaders(f: PartialFunction[Header, Header]): Middleware[Any, Nothing]
```

Use this constructor to implement signCookies.